mobile/lib/shared/services/sync.service.dart

```diff
@@ -130,13 +138,46 @@ class SyncService {
     return true;
   }
 
-  /// Syncs remote assets to the databas
-  /// returns `true` if there were any changes
-  Future<bool> _syncRemoteAssetsToDb(
+  /// Efficiently syncs assets via changes. Returns `null` when a full sync is required.
+  Future<bool?> _syncRemoteAssetChanges(
+    User user,
+    Future<(List<Asset>? toUpsert, List<String>? toDelete)> Function(
+      User user,
+      DateTime since,
+    ) getChangedAssets,
+  ) async {
+    final DateTime? since = _db.eTags.getByIdSync(user.id)?.time?.toUtc();
+    if (since == null) return null;
+    final DateTime now = DateTime.now();
+    final (toUpsert, toDelete) = await getChangedAssets(user, since);
+    if (toUpsert == null || toDelete == null) return null;
+    try {
+      if (toDelete.isNotEmpty) {
+        await _db.writeTxn(() => _db.assets.deleteAllByRemoteId(toDelete));
```

Comment:
I'll need to change this to use the existing delete logic that accounts for local assets staying in DB even if the remote part is removed